src/modules/75-cf/utils/__tests__/UsageLimitUtils.test.ts

```diff
@@ -0,0 +1,165 @@
+/*
+ * Copyright 2022 Harness Inc. All rights reserved.
+ * Use of this source code is governed by the PolyForm Shield 1.0.0 license
+ * that can be found in the licenses directory at the root of this repository, also available at
+ * https://polyformproject.org/wp-content/uploads/2020/06/PolyForm-Shield-1.0.0.txt.
+ */
+
+import { BannerType } from '@common/layouts/Constants'
+import { getBannerText } from '../UsageLimitUtils'
+
+describe('Usage Limit Utils', () => {
+  const limit = 25000
+  describe('when user has the Free Plan', () => {
+    const additionalLicenseProps = {
+      isFreeEdition: true,
+      isEnterpriseEdition: false,
+      isTeamEdition: false
+    }
+
+    test('it should not display message or banner when usage is below 90%', () => {
+      const count = 22000
+      const expectedMessage = undefined
+      const getString = jest.fn().mockReturnValue(expectedMessage)
+
+      const { message, bannerType } = getBannerText(getString, additionalLicenseProps, count, limit)
+
+      expect(message).toEqual(expectedMessage)
+      expect(bannerType).toEqual(undefined)
+    })
+
+    test('it should display message and banner when usage is 90%', () => {
+      const count = 22500
+      const expectedMessage = 'You have used 90% of your Monthly Active Users (MAU) subscription limit.'
+      const getString = jest.fn().mockReturnValue(expectedMessage)
+
+      const { message, bannerType } = getBannerText(getString, additionalLicenseProps, count, limit)
+
+      expect(message).toEqual(expectedMessage)
+      expect(bannerType).toEqual(BannerType.INFO)
+    })
+
+    test('it should return correct message when usage is above 90% and less than 100%', () => {
+      const count = 24000
+      const expectedMessage =
+        'You have used 96% of Monthly Active Users (MAU) included in the free plan. After 25K MAUs, flag management will be restricted.'
+
+      const getString = jest.fn().mockReturnValue(expectedMessage)
+      const { message, bannerType } = getBannerText(getString, additionalLicenseProps, count, limit)
+
+      expect(message).toEqual(expectedMessage)
+      expect(bannerType).toEqual(BannerType.INFO)
+    })
+
+    test('it should return correct message when usage is above 99%', () => {
+      const count = 24999
+
+      const expectedMessage =
+        'You have used 0.96% of Monthly Active Users (MAU) included in the free plan. After 25K MAUs, flag management will be restricted.'
```

Comment:
This message is a bit misleading, as we're not expecting this

```suggestion
        'You have used 99% of Monthly Active Users (MAU) included in the free plan. After 25K MAUs, flag management will be restricted.'
```